try {
    timeout(time: 20, unit: 'MINUTES') {
        node('nodejs') {
            stage('build') {
            openshift.withCluster() {
                openshift.withProject() {
                    echo 'Building..'
                }  
            }
            }
            stage('deploy') {
            openshift.withCluster() {
                openshift.withProject() {
                echo 'Deploying ..'
                }
            }
            }
        }
    }
} catch (err) {
    echo "in catch block"
    echo "Caught: ${err}"
    currentBuild.result = 'FAILURE'
    throw err
}          